### PR TITLE
updating k8s example for use in blog post

### PR DESCRIPTION
**Pull request by puja108**
first commit:
based on releaseit output
v1.0.1 currently
still about to move images to public dockerhub

### CODE REVIEW

**Comment by puja108** [comment]
Works well with 1.0.1, 1.0.2, 1.2.0 and most probably even 1.2.2 (I didn't upload any other k8s version to dockerhub). Goal is to show 1.2.0 and then a rolling update of the nodes to 1.2.2. For now without any scheduled pods, but might add a tiny pod in there so you can see that it survives (might need to tweak wait time a bit).

**Comment by puja108** [comment]
More todos:
- [ ] unit files might need some cleaning up
- [x] not sure if `setup-network-env` needs to be in the master group or should be by itself (@teemow ?)

**Comment by puja108** [comment]
This is now working incl rolling updates with a running (scaled) pod and in line with the soon to be published inago blog post. Should be clean enough for a first publication.

RFR @giantswarm/cena

File: docs/k8s.md [hunk 1]
```diff
@@ -1,47 +1,151 @@
...
+
+Now, we can schedule the nodes. As we have a 3 node cluster we'll be running 3 slices of nodes:
+
+```
+$ inagoctl up k8s-node 3
+2016-04-14 15:41:26.824 | INFO     | context.Background: Succeeded to submit group 'k8s-node'.
+2016-04-14 15:41:36.847 | INFO     | context.Background: Succeeded to start 3 slices for group 'k8s-node'
+```
+
+And we're done!
```
Comment (xh3b4sd) [inline docs/k8s.md:45]: I am wondering why we need three different groups. I want to execute one command and magic should happen. What are we missing to do this? Why do we need three commands? One would be super awesome.

File: docs/k8s.md [hunk 2]
```diff
@@ -1,47 +1,151 @@
...
+
+Looking closely at the update command above we can inspect some of its options. `--max-growth=0` tells Inago to not start any additional instances, which means it has to remove an instance to be able to start a new one. In this example this is due to the fact that we don't have any free machine left. `--min-alive=2` tells Inago to keep at least two instances of the group alive during the update process. the `ready-secs` flag determines how long Inago waits between rounds of updating. This gives the new node a bit of time to start replicating that pod we scheduled.
+
+Watching `kubectl get deployments` during the update process, we would see that available instances of our hello-node pod shortly go down to 2, while a Kubernetes node is being updated and then quickly come up again to 3.
+
+### Updating the Master
+
+Now sadly, we didn't use an HA deployment of the Kubernetes master, so Inago's update command isn't of help much here. Accordingly, we will have a short downtime of our API while we update this group. Luckily, Kubernetes nodes and their pods don't rely on the API server that much and will usually stay running while we update the master.
```
Comment (xh3b4sd) [inline docs/k8s.md]: `isn't of help` is weird.

File: docs/k8s.md [hunk 3]
```diff
@@ -1,47 +1,151 @@
...
+
+$ inagoctl destroy k8s-master
+2016-04-14 16:32:27.384 | INFO     | context.Background: Succeeded to destroy group 'k8s-master'.
+
+$ inagoctl up k8s-master
+2016-04-14 16:35:23.503 | INFO     | context.Background: Succeeded to submit group 'k8s-master'.
+2016-04-14 16:36:30.600 | INFO     | context.Background: Succeeded to start group 'k8s-master'.
 ```
+
+A quick look at `kubectl version` will tell us that we're now running a 1.2.2 server.
```
Comment (xh3b4sd) [inline docs/k8s.md]: Based on a gut feeling I miss some "ending words". Here the post is suddenly over and I have the feeling something is missing. Nobody said it is over. Nobody said goodby. Does this make sense?

**Comment by xh3b4sd** [comment]
Minor comments. The post is cool. I only still don't like the different locations of docs and unit files.

File: docs/k8s.md (hunk 1 above)
Comment (puja108) [inline docs/k8s.md:45]: The network one can be integrated with the master, but I thought it's cleaner outside as it's one-off and actually does not belong to k8s itself.

master and nodes need to be deployed separately as the master in this example is not HA (and not sliced).

Also, you would want some control over master separately from nodes as they are two different things. Even with an HA mster you would maybe want only 3 slices of the master, but you might want 10 nodes.

File: docs/k8s.md (hunk 2 above)
Comment (puja108) [inline docs/k8s.md]: true, changed to `currently won't work`

File: docs/k8s.md (hunk 3 above)
Comment (puja108) [inline docs/k8s.md]: yes, I agree, actually have some in the blog post, but kinda forgot them here. will add.

File: docs/k8s.md (hunk 1 above)
Comment (xh3b4sd) [inline docs/k8s.md:45]: I see the group differentiation makes sense. Anyway I think even for multiple groups one `up` should be enough. Having that said there would be a need for group specific configuration. Then it starts to feel not that handy anymore.

File: docs/k8s.md [hunk 4]
```diff
@@ -1,47 +1,151 @@
 # Kubernetes
 
 With Inago you can easily deploy Kubernetes on a CoreOS cluster.
-The `example` folder contains a `k8s` example. Kubernetes is split into
+The [example folder](https://github.com/giantswarm/inago/tree/master/example) contains a simple [k8s example](https://github.com/giantswarm/inago/tree/master/example/k8s). The Kubernetes deployment is split into
 two groups: `k8s-master` and `k8s-node`.
 
 ## Note
 The deployment sets up a Kubernetes cluster, which uses insecure communication
```
Comment (zeisss) [inline docs/k8s.md]: I would change this `which`.

> The deployment sets up a Kubernetes cluster with insecure communication between the components.

File: docs/k8s.md [hunk 5]
```diff
@@ -1,47 +1,151 @@
...
 between the components. Use this only as a Demo or POC. If you want to deploy a
 Kubernetes cluster for production please modify the example, so secure
-communication is used. [Cluster TLS using OpenSSL](https://coreos.com/Kubernetes/docs/latest/openssl.html)
+communication is used: [Cluster TLS using OpenSSL](https://coreos.com/Kubernetes/docs/latest/openssl.html).
+Further, this deployment uses a single Kubernetes master, so the API is not highly available. For HA deployments of Kubernetes check out: [Building High-Availability Clusters](http://kubernetes.io/docs/admin/high-availability/).
```
Comment (zeisss) [inline docs/k8s.md]: I wouldn't put a double colon here.

File: docs/k8s.md (hunk 4 above)
Comment (puja108) [inline docs/k8s.md]: done

File: docs/k8s.md (hunk 5 above)
Comment (puja108) [inline docs/k8s.md]: ack

**Comment by puja108** [comment]
@xh3b4sd is my answer to your 1 vs many groups comment acceptable, if LGTY

**Comment by puja108** [comment]
fixes https://github.com/giantswarm/inago/issues/220

**Comment by xh3b4sd** [comment]
Yeah, discussed. LGTM

**Pull request merged**